### PR TITLE
JAMES-2334 Demonstrate that published messages are not lost without consumer

### PR DESCRIPTION
**Pull request by aduprat**
Only last commit

### CODE REVIEW

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        try (Connection connection = connectionFactory.newConnection();
+                Channel channel = connection.createChannel();) {
+            String exchangeName = "exchangeName";
+            String routingKey = "routingKey";
+            String queueName = createQueue(channel, exchangeName, routingKey);
+
+            publishAMessage(channel, exchangeName, routingKey);
+
+            Thread.sleep(ONE_MINUTE);
+            boolean autoAck = false;
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: Could be a constant?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        connectionFactory.setUsername(rabbitMQ.getUsername());
+        connectionFactory.setPassword(rabbitMQ.getPassword());
+    }
+
+    @Test
+    public void publishedEventWithoutSubscriberShouldNotBeLost() throws Exception {
+        try (Connection connection = connectionFactory.newConnection();
+                Channel channel = connection.createChannel();) {
+            String exchangeName = "exchangeName";
+            String routingKey = "routingKey";
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: These strings can be constants...

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            publishAMessage(channel, exchangeName, routingKey);
+
+            Thread.sleep(ONE_MINUTE);
+            boolean autoAck = false;
+            assertThat(channel.basicGet(queueName, autoAck)).isNotNull();
+        }
+    }
+
+    private String createQueue(Channel channel, String exchangeName, String routingKey) throws IOException {
+        boolean durable = true;
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: Constant?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        channel.exchangeDeclare(exchangeName, "direct", durable);
+        String queueName = channel.queueDeclare().getQueue();
+        channel.queueBind(queueName, exchangeName, routingKey);
+        return queueName;
+    }
+
+    private void publishAMessage(Channel channel, String exchangeName, String routingKey) throws IOException {
+        byte[] messageBodyBytes = "Hello, world!".getBytes();
+        BasicProperties properties = null;
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: Constant?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java (hunk 1 above)
Comment (aduprat) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: I'm not really fan of introducing a lot of constant, only if they are used twice; else we don't gain that much.

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java (hunk 2 above)
Comment (aduprat) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: same

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java (hunk 3 above)
Comment (aduprat) [reply server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: same

**Comment by chibenwa** [comment]
test this please

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        connectionFactory.setHost(rabbitMQ.getHostIp());
+        connectionFactory.setPort(rabbitMQ.getPort());
+        connectionFactory.setUsername(rabbitMQ.getUsername());
+        connectionFactory.setPassword(rabbitMQ.getPassword());
+    }
+
+    @Test
+    public void publishedEventWithoutSubscriberShouldNotBeLost() throws Exception {
+        try (Connection connection = connectionFactory.newConnection();
+                Channel channel = connection.createChannel();) {
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: Still extra ';'  here

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+@ExtendWith(DockerRabbitMQExtension.class)
+public class RabbitMQTest {
+
+    private static final long ONE_MINUTE = TimeUnit.MINUTES.toMillis(1);
+    private ConnectionFactory connectionFactory;
+
+    @BeforeEach
+    public void setup(DockerRabbitMQ rabbitMQ) {
+        connectionFactory = new ConnectionFactory();
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: you should have a rabbitMQ.connectionFactory() for that

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    public void publishedEventWithoutSubscriberShouldNotBeLost() throws Exception {
+        try (Connection connection = connectionFactory.newConnection();
+                Channel channel = connection.createChannel();) {
+            String exchangeName = "exchangeName";
+            String routingKey = "routingKey";
+            String queueName = createQueue(channel, exchangeName, routingKey);
+
+            publishAMessage(channel, exchangeName, routingKey);
+
+            Thread.sleep(ONE_MINUTE);
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: are you serious, a test with a one minute sleep ?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    private String createQueue(Channel channel, String exchangeName, String routingKey) throws IOException {
+        boolean durable = true;
+        channel.exchangeDeclare(exchangeName, "direct", durable);
+        String queueName = channel.queueDeclare().getQueue();
+        channel.queueBind(queueName, exchangeName, routingKey);
+        return queueName;
+    }
+
+    private void publishAMessage(Channel channel, String exchangeName, String routingKey) throws IOException {
+        byte[] messageBodyBytes = "Hello, world!".getBytes();
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQTest.java]: encoding please

**Comment by mbaechler** [comment]
merged in another PR

**Pull request closed without merging**